ynput/ayon-applications
# Chore: Applications settings

## Changelog Description
Refactored applications settings. Removed unnecessary settings from known applications like icon and label. Added option to define available applications and tools in settings instead of attributes.

## Additional info
Using settings instead of attributes for available applications and tools is for now by default turned off, that is to avoid possible confusion on update and to keep opened doors if we find out possible issues with current settings.

The possible issues might be that tools cannot be set per task, but only per folder type and task type, which is more less granular approach, but at the same time easier to set up.

Removed settings for icon and label, known applications have both hardcoded in code.

### Screenshot
![image](https://github.com/user-attachments/assets/095d010c-c2f9-46ed-9fd8-129a7851e2e1)

### UX enhancements
~~I would say that the labels in settings might need a change. As there is a lot of places where `Applications` and `Tools` is visible in UI.~~

~~Also I've used labels from settings labels in launcher, which might be wrong? (e.g. `Maya` became `Autodesk Maya`.)~~ Removed company name from application labels.

Tools now have filtering by application and host name in tools definitions. It might be also added to profiles. But I think that current model is better, as the tool define application and host name, whereas tools profiles define context where it will be used, maybe?

NOTE: This PR requires minor version bump.

## Testing notes:
1. ~~Change version in `package.py` to higher minor version (e.g. `0.3.0-settings`).~~ Version in PR already changed to `1.0.0-dev.1` .
2. Run `python ./create_package.py` and upload the package to AYON server.
3. Use the addon in a bundle and set the bundle as production (or current dev).
4. With that done, you should see applications in launcher as before, using attributes.
5. Copy settings from older version of applications addon.
6. Now you should have `ayon+settings://applications/project_applications/enabled` and `ayon+settings://applications/project_tools/enabled` set to `False` They are both enabled by default, but copying of settings from older version should change it. Change it to `True`, so you can test new features.
7. Create new profile in `ayon+settings://applications/project_applications` and define applications that will be shown. You can add different profiles for different task types or folder types.
8. Launcher should show applications based on the settings profiles.
9. Create new profile for tools in `ayon+settings://applications/project_tools` and define tools that should be used.
10. Launch application and validate that correct tools were used based on the settings profiles.
